src/hooks/cakeVault/useGetVaultBountyInfo.ts

```diff
@@ -2,44 +2,38 @@ import { useState, useEffect } from 'react'
 import BigNumber from 'bignumber.js'
 import { useGetApiPrice } from 'state/hooks'
 import { useCakeVaultContract } from 'hooks/useContract'
-import { getFullDisplayBalance } from 'utils/formatBalance'
+import useRefresh from 'hooks/useRefresh'
 import makeBatchRequest from 'utils/makeBatchRequest'
 import { getCakeAddress } from 'utils/addressHelpers'
 
-const useGetVaultBountyInfo = (refresh?: number) => {
+const useGetVaultBountyInfo = () => {
+  const { fastRefresh } = useRefresh()
   const cakeVaultContract = useCakeVaultContract()
-  const [estimatedCallBountyReward, setEstimatedCallBountyReward] = useState(null)
-  const [totalPendingCakeRewards, setTotalPendingCakeRewards] = useState(null)
-  const [dollarCallBountyToDisplay, setDollarBountyToDisplay] = useState(null)
-  const [cakeCallBountyToDisplay, setCakeBountyToDisplay] = useState(null)
+  const [estimatedDollarBountyReward, setEstimatedDollarBountyReward] = useState(null)
+  const [estimatedCakeBountyReward, setEstimatedCakeBountyReward] = useState(null)
+  const [totalPendingCakeHarvest, setTotalPendingCakeHarvest] = useState(null)
 
   const cakePrice = useGetApiPrice(getCakeAddress())
 
   useEffect(() => {
-    // Call contract to get estimated rewards
     const fetchRewards = async () => {
-      const [estimatedRewards, pendingCakeRewards] = await makeBatchRequest([
+      const [estimatedClaimableCakeReward, pendingTotalCakeHarvest] = await makeBatchRequest([
         cakeVaultContract.methods.calculateHarvestCakeRewards().call,
         cakeVaultContract.methods.calculateTotalPendingCakeRewards().call,
       ])
-      setEstimatedCallBountyReward(new BigNumber(estimatedRewards as string))
-      setTotalPendingCakeRewards(new BigNumber(pendingCakeRewards as string))
+      if (cakePrice) {
+        const dollarValueOfClaimableReward = new BigNumber(estimatedClaimableCakeReward as string).multipliedBy(
+          cakePrice,
+        )
+        setEstimatedDollarBountyReward(dollarValueOfClaimableReward)
+      }
+      setEstimatedCakeBountyReward(new BigNumber(estimatedClaimableCakeReward as string))
+      setTotalPendingCakeHarvest(new BigNumber(pendingTotalCakeHarvest as string))
     }
     fetchRewards()
-  }, [cakeVaultContract, refresh])
+  }, [cakeVaultContract, cakePrice, fastRefresh])
 
-  useEffect(() => {
-    // Convert estimated rewards to dollars and a cake display value
-    if (estimatedCallBountyReward && cakePrice) {
-      const dollarValueOfReward = estimatedCallBountyReward.multipliedBy(cakePrice)
-      const estimatedDollars = getFullDisplayBalance(dollarValueOfReward, 18, 2)
-      const estimatedCake = getFullDisplayBalance(estimatedCallBountyReward, 18, 3)
-      setDollarBountyToDisplay(estimatedDollars)
-      setCakeBountyToDisplay(estimatedCake)
-    }
-  }, [cakePrice, estimatedCallBountyReward])
```

Comment:
Moving all of the `getFullDisplayBalance` logic out of the hook and into the card. The hook returns the raw values, the component formats the data as it needs.